app/controllers/api/project.php

```diff
@@ -110,3 +118,198 @@
 
         $response->dynamic($usage, Response::MODEL_USAGE_PROJECT);
     });
+
+
+// Variables
+
+App::post('/v1/project/variables')
+    ->desc('Create Variable')
+    ->groups(['api'])
+    ->label('scope', 'projects.write')
+    ->label('audits.event', 'variable.create')
+    ->label('sdk.auth', [APP_AUTH_TYPE_KEY])
+    ->label('sdk.namespace', 'project')
+    ->label('sdk.method', 'createVariable')
+    ->label('sdk.description', '/docs/references/project/create-variable.md')
+    ->label('sdk.response.code', Response::STATUS_CODE_CREATED)
+    ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
+    ->label('sdk.response.model', Response::MODEL_VARIABLE)
+    ->param('key', null, new Text(Database::LENGTH_KEY), 'Variable key. Max length: ' . Database::LENGTH_KEY  . ' chars.', false)
+    ->param('value', null, new Text(8192), 'Variable value. Max length: 8192 chars.', false)
+    ->inject('project')
+    ->inject('response')
+    ->inject('dbForProject')
+    ->action(function (string $key, string $value, Document $project, Response $response, Database $dbForProject) {
+        $variableId = ID::unique();
+
+        $variable = new Document([
+            '$id' => $variableId,
+            '$permissions' => [
+                Permission::read(Role::any()),
+                Permission::update(Role::any()),
+                Permission::delete(Role::any()),
+            ],
+            'resourceInternalId' => '',
+            'resourceId' => '',
+            'resourceType' => 'project',
+            'key' => $key,
+            'value' => $value,
+            'search' => implode(' ', [$variableId, $key, 'project']),
```

Comment:
Do we need a fulltext field, since the attributes : $variableId, $key, 'project' have independent fields we can search on?

Reply:
Yes, because we have a generic search, this is not specific to Functions G4, can be tackled later.